cloudcix_primitives/cloudinit_kvm.py

```diff
@@ -0,0 +1,774 @@
+        3426: f'Failed to read domain {domain} state from host {host}',
+        3427: f'Failed to connect to the host {host} for payload destroy_domain',
+        3428: f'Failed to destroy domain {domain} on host {host}',
+    }
+
+    def run_host(host, prefix, successful_payloads):
+        rcc = SSHCommsWrapper(comms_ssh, host, 'robot')
+        fmt = HostErrorFormatter(
+            host,
+            {'payload_message': 'STDOUT', 'payload_error': 'STDERR'},
+            successful_payloads
+        )
+
+        payloads = {
+            'read_domstate_0': f'virsh domstate {domain} ',
+            'shutdown_domain': f'virsh shutdown {domain} ',
+            'read_domstate_n': f'virsh domstate {domain} ',
+            'destroy_domain': f'virsh destroy {domain} ',  # force shutdown = destroy
+        }
+
+        # first read the state before shutdown the domain
+        ret = rcc.run(payloads['read_domstate_0'])
+        if ret["channel_code"] != CHANNEL_SUCCESS:
+            return False, fmt.channel_error(ret, f'{prefix + 1}: {messages[prefix + 1]}'), fmt.successful_payloads
+        quiesced = False
+        if ret["payload_code"] != SUCCESS_CODE:
+            return False, fmt.payload_error(ret, f'{prefix + 2}: {messages[prefix + 2]}'), fmt.successful_payloads
+        else:
+            if 'shut off' in ret["payload_message"].strip():
+                quiesced = True
+        fmt.add_successful('read_domstate_0', ret)
+
+        if quiesced is True:
+            return True, "", fmt.successful_payloads
+
+        ret = rcc.run(payloads['shutdown_domain'])
+        if ret["channel_code"] != CHANNEL_SUCCESS:
+            return False, fmt.channel_error(ret, f'{prefix + 3}: {messages[prefix + 3]}'), fmt.successful_payloads
+        if ret["payload_code"] != SUCCESS_CODE:
+            return False, fmt.payload_error(ret, f'{prefix + 4}: {messages[prefix + 4]}'), fmt.successful_payloads
+        fmt.add_successful('shutdown_domain', ret)
+
+        # Since shutdown is run make sure it is in shutoff state, so read the state until it is shutoff
+        # for max 300 seconds
+        start = 0
+        shutoff = False
+        while start < 300 and shutoff is False:
+            ret = rcc.run(payloads['read_domstate_n'])
+            if ret["channel_code"] != CHANNEL_SUCCESS:
+                fmt.channel_error(
+                    ret, f'Attempt at {start} seconds: {prefix + 5}: {messages[prefix + 5]}'
+                ), fmt.successful_payloads
+            if ret["payload_code"] != SUCCESS_CODE:
+                fmt.payload_error(
+                    ret, f'Attempt at {start} seconds: {prefix + 6}: {messages[prefix + 6]}'
+                ), fmt.successful_payloads
+            else:
+                if 'shut off' in ret["payload_message"].strip():
+                    shutoff = True
+                else:
+                    # wait interval is 0.5 seconds
+                    time.sleep(0.5)
+                    start += 0.5
```

Comment:
This is not accounting for the time for the loop to run. It equates to looping a max of 600 times but is not reflective in time for sending payload, getting response etc.